### PR TITLE
Auto hide the scrollbar by default and Added ` linenum_border` 

### PR DESCRIPTION
**Pull request by Sidmaz666**
1.  Set Scrollbar Auto hide to True by Default.
2.  Added new param for CodeView, `linenum_border`  to specify the border width of the Line Number Canvas.

### CODE REVIEW

**Review by whmsft — APPROVED** [review]
Looks good enough to me... Let the "Benevolent Dictator" @rdbende review it himself

**Comment by Moosems** [comment]
Woah @whmsft, haven't seen you in a hot minute!

**Comment by whmsft** [comment]
> Woah @whmsft, haven't seen you in a hot minute!

Great to reconnect dear. I’ve been into tiny game development lately, but my studies have kept me busy... 
Hope you’re well! 😊

File: chlorophyll/codeview.py [hunk 1]
```diff
@@ -46,7 +46,8 @@ def __init__(
         color_scheme: dict[str, dict[str, str | int]] | str | None = None,
         tab_width: int = 4,
         linenums_theme: Callable[[], tuple[str, str]] | tuple[str, str] | None = None,
-        autohide_scrollbar: bool = False,
+        autohide_scrollbar: bool = True,
+        linenum_border = 0,
```
Comment (rdbende) [inline chlorophyll/codeview.py:50]:
```suggestion
        linenums_border: int = 0,
```

File: chlorophyll/codeview.py [hunk 2]
```diff
@@ -60,7 +61,11 @@ def __init__(
         super().grid(row=0, column=1, sticky="nswe")
 
         self._line_numbers = TkLineNumbers(
-            self._frame, self, justify=kwargs.get("justify", "left"), colors=linenums_theme
+            self._frame,
+            self,
+            justify=kwargs.get("justify", "left"),
+            colors=linenums_theme,
+            borderwidth=kwargs.get("borderwidth", linenum_border)
```
Comment (rdbende) [inline chlorophyll/codeview.py:68]:
```suggestion
            borderwidth=kwargs.get("borderwidth", linenums_border)
```

**Review by rdbende — CHANGES_REQUESTED** [review]
I'm not sure you wanted to commit `example.py`, or it's just a mistake, but I'd rather not have an example file in the project root directory.

File: README.md [hunk 3]
```diff
@@ -29,6 +29,7 @@ pip install -e .
 |color_scheme      |A color scheme for the code     |Dict, string, or toml file                    |
 |tab_width         |The width of a tab (`\t`)       |Int                                           |
 |autohide_scrollbar|Auto hide scrollbars            |Bool                                          |
+|linenum_border    |Line Number Border Width        |Int                                           |
```
Comment (rdbende) [inline README.md:32]:
```suggestion
|linenums_border   |Border width of the line numbers|Int                                           |
```

Comment (rdbende) [reply README.md:32]: Hmm, actually `linenums_theme` is missing from here @Moosems

**Comment by Moosems** [comment]
Honestly kinda sad I'm getting left out of the reviews when I wrote the linebar lib 🤣

**Comment by rdbende** [comment]
@Moosems, please review and squash merge it, if it looks good to you :)

**Review by Moosems — COMMENTED** [review]
The code all checks out by why are we auto hiding the scroll bar by default?

**Comment by Moosems** [comment]
I just see it causing issues for those who have it in older projects and never used the feature and them being confused why their scroll bar is gone now

File: chlorophyll/codeview.py [hunk 4]
```diff
@@ -46,7 +46,8 @@ def __init__(
         color_scheme: dict[str, dict[str, str | int]] | str | None = None,
         tab_width: int = 4,
         linenums_theme: Callable[[], tuple[str, str]] | tuple[str, str] | None = None,
-        autohide_scrollbar: bool = False,
+        autohide_scrollbar: bool = True,
```
[the remark this answers is not in the document]
Comment (rdbende) [reply chlorophyll/codeview.py:49]: Because there's really no reason for having a scrollbar when the view isn't scrollable.

**Comment by rdbende** [comment]
> I just see it causing issues for those who have it in older projects and never used the feature and them being confused why their scroll bar is gone now

Then I think we can just mark it as a breaking change (even though it isn't) and release it as 0.5

File: chlorophyll/codeview.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Moosems) [reply chlorophyll/codeview.py:49]: The issue is that it forces resizing which can become a complication

[the remark this answers is not in the document]
Comment (rdbende) [reply chlorophyll/codeview.py:49]: That's something that should be fixed regardless, but making this behaviour default is sensible to me.

**Review by rdbende — APPROVED** [review]

**Pull request merged**